### PR TITLE
chore: Bump rust version to 1.80

### PR DESCRIPTION
**Pull request by DSharifi**
This PR bumps the rust compiler to version 1.80.

Changes:
- Fixed rust docs to have correct indentation.
- Added missing crate features that were defined in `BUILD.Bazel` but missing in `Cargo.toml`.
- Add optional dependencies in `Cargo.toml` files required to build fuzzer.
- Removed dead code
- Removed allow clippy warnings that were previously false positives.
- Removed code that were feature gated with undefined features.

### CODE REVIEW

**Review by NikolasHai — APPROVED** [review]

**Review by rumenov — APPROVED** [review]

**Review by dist1ll — APPROVED** [review]

File: rs/consensus/src/consensus.rs [hunk 1]
```diff
@@ -374,23 +374,23 @@ impl<T: ConsensusPool> ChangeSetProducer<T> for ConsensusImpl {
...
     ///
     /// 1. It must return immediately if one of the subcomponent returns a
-    /// non-empty [ChangeSet]. It is important that a [ChangeSet] is fully
-    /// applied to the pool or timer before another subcomponent uses
-    /// them, because each subcomponent expects to see full state in order to
-    /// make correct decisions on what to do next.
+    ///    non-empty [ChangeSet]. It is important that a [ChangeSet] is fully
+    ///    applied to the pool or timer before another subcomponent uses
+    ///    them, because each subcomponent expects to see full state in order to
+    ///    make correct decisions on what to do next.
```
Comment (dist1ll) [inline rs/consensus/src/consensus.rs:380]: Was that done by the Rust formatter or clippy?

**Review by dist1ll — APPROVED** [review]
LGTM for consensus code

**Review by mraszyk — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

File: rs/bitcoin/consensus/src/payload_builder.rs [hunk 2]
```diff
@@ -4,8 +4,10 @@ mod parse;
 #[cfg(test)]
 mod tests;
 
-#[cfg(all(test, feature = "proptest"))]
-mod proptests;
+// TODO: This module does not compile, and was not caught by the compiler
+// as it was behind a non-existent feature flag.
+// #[cfg(test)]
+// mod proptests;
```
Comment (dsarlis) [inline rs/bitcoin/consensus/src/payload_builder.rs:10]: Thanks for catching this. I can fix it in a follow-up.

**Review by schneiderstefan — APPROVED** [review]

**Review by altkdf — APPROVED** [review]

File: rs/boundary_node/denylist_updater/src/main.rs [hunk 3]
```diff
@@ -474,30 +470,6 @@ impl<T: List> List for WithNormalize<T> {
     }
 }
 
-struct WithRecover<T: List>(T);
-
-#[async_trait]
-impl<T: List> List for WithRecover<T> {
```
Comment (nikolay-komarevskiy) [inline rs/boundary_node/denylist_updater/src/main.rs:480]: @DSharifi You deleted it because it is unused?

Comment (DSharifi) [reply rs/boundary_node/denylist_updater/src/main.rs:480]: Yes

File: rs/embedders/src/wasmtime_embedder.rs [hunk 4]
```diff
@@ -1,3 +1,8 @@
+// False positive clippy lint.
+// Issue: https://github.com/rust-lang/rust-clippy/issues/12856
+// Fixed in: https://github.com/rust-lang/rust-clippy/pull/12892
+#![allow(clippy::needless_borrows_for_generic_args)]
```
Comment (adambratschikaye) [inline rs/embedders/src/wasmtime_embedder.rs:4]: Could we make a Jira item to remove this allow (and the two other cases under `rs/embedders`) on the next rustc version bump? I just don't want us to forget it.

**Review by adambratschikaye — APPROVED** [review]

File: rs/consensus/src/consensus.rs (hunk 1 above)
Comment (DSharifi) [reply rs/consensus/src/consensus.rs:380]: Neither. It was done manually :') to satisfy a new lint introduced to clippy.

https://rust-lang.github.io/rust-clippy/master/index.html#/doc_lazy_continuation

File: rs/embedders/src/wasmtime_embedder.rs (hunk 4 above)
Comment (DSharifi) [reply rs/embedders/src/wasmtime_embedder.rs:4]: I have it noted down for the next upgrade, but created a JIRA ticket for it as well now. NET-1737

**Review by nikolay-komarevskiy — APPROVED** [review]

File: rs/boundary_node/ic_boundary/src/acme.rs [hunk 5]
```diff
@@ -33,6 +33,7 @@ pub trait Ready: Sync + Send {
 
 #[derive(Debug, thiserror::Error)]
 pub enum FinalizeError {
+    #[allow(dead_code)]
```
Comment (nikolay-komarevskiy) [inline rs/boundary_node/ic_boundary/src/acme.rs:36]: Why is this needed? From the first glance it looks like the code is [used](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/boundary_node/ic_boundary/src/acme.rs?L141)?

Comment (DSharifi) [reply rs/boundary_node/ic_boundary/src/acme.rs:36]: There's a lot of `#[allow(dead_code)]` in `/rs/boundary_node` for code that is actually used. Cargo clippy complains that the code is dead, and I assume it's due to some feature gates that aren't correctly configured in `Cargo.toml`. But I am not sure, and haven't been able to figure out the root cause.

Comment (nikolay-komarevskiy) [reply rs/boundary_node/ic_boundary/src/acme.rs:36]: Gotcha, let's fix it on the proper level before merging, i could help later (got sick now)

File: rs/nervous_system/common/build_metadata/src/lib.rs [hunk 6]
```diff
@@ -27,13 +27,13 @@ const BUILD_INFO_FORMAT: &str = r#"
...
-///     d. dfn_core
-///   2. List build-info-build in the [build-dependencies] section of their Cargo.toml.
-///   3. Have a ./build.rs where main calls build_info::build_script() (either directly or indirectly).
+/// 1. List the following in the [dependencies] section of their Cargo.toml:
+///     * build-info
+///     * candid
+///     * dfn_candid
+///     * dfn_core
+/// 2. List build-info-build in the [build-dependencies] section of their Cargo.toml.
+/// 3. Have a ./build.rs where main calls build_info::build_script() (either directly or indirectly).
```
Comment (daniel-wong-dfinity-org) [inline rs/nervous_system/common/build_metadata/src/lib.rs:36]: This is tangential.

Also, I do not see how this is an improvement. Seems like moving deck chairs on the Titanic.

Also, do you really want to show up in git blame?

At first, I was going to say "Do as you see fit", but as I look at more of this PR, I see that there is a fair amount of comment reformatting. Because the amount of such changes is "substantial", I'm thinking that we should rethink these changes. Maybe I'm missing some context for why you are making these changes in this PR?

File: rs/nns/governance/src/neuron_store.rs [hunk 7]
```diff
@@ -105,11 +105,6 @@ impl NeuronStoreError {
     }
 }
 
-#[derive(Debug, Eq, PartialEq)]
-pub struct NeuronNotFound {
-    neuron_id: NeuronId,
-}
-
```
Comment (daniel-wong-dfinity-org) [inline rs/nns/governance/src/neuron_store.rs:112]: Is this causing a warning or some other problem? Otherwise, please, revert.

File: rs/nns/governance/tests/interleaving/mod.rs
Comment (daniel-wong-dfinity-org) [inline rs/nns/governance/tests/interleaving/mod.rs:1]: The changes here are pretty substantial; therefore, they probably should not be entrained in a PR whose primary purpose is to upgrade Rust.

File: rs/nns/gtc/canister/canister.rs [hunk 8]
```diff
@@ -179,7 +179,6 @@ fn forward_whitelisted_unclaimed_accounts() {
 }
 
 #[candid_method(update, rename = "forward_whitelisted_unclaimed_accounts")]
-#[allow(clippy::let_unit_value)] // clippy false positive
```
Comment (daniel-wong-dfinity-org) [inline rs/nns/gtc/canister/canister.rs:182]: Entraining this delete is fine even though it is tangential, because a) it is super small b) it is a substantive improvement, not just changing comment formatting c) it won't show up in blame, since it is an outright delete, not a modification.

**Review by daniel-wong-dfinity-org — CHANGES_REQUESTED** [review]

File: rs/nns/governance/src/neuron_store.rs (hunk 7 above)
Comment (DSharifi) [reply rs/nns/governance/src/neuron_store.rs:112]: It is dead code :)

Clippy rightly gives a warning making CI fail.

File: rs/nervous_system/common/build_metadata/src/lib.rs (hunk 6 above)
Comment (DSharifi) [reply rs/nervous_system/common/build_metadata/src/lib.rs:36]: The changes are needed as the new clippy version detects that the generated MD can be broken,  [doc_lazy_continuation](https://rust-lang.github.io/rust-clippy/master/index.html#/doc_lazy_continuation). 

No, I don't mind that the changes show up in the git blame history, as it was part of what's needed to upgrade the compiler version.

File: rs/nns/gtc/canister/canister.rs (hunk 8 above)
Comment (DSharifi) [reply rs/nns/gtc/canister/canister.rs:182]: I added it in the previous rust version bump as there was a bug in clippy. We delete it now given that the false positive case here is no longer triggered in the upgraded clippy version.

The idea is to delete them in subsequent upgrades when the bug is fixed and the allow statement is no longer needed

See https://github.com/dfinity/ic/pull/642#discussion_r1694758879

**Review by rikonor — APPROVED** [review]

File: rs/nns/gtc/canister/canister.rs (hunk 8 above)
Comment (daniel-wong-dfinity-org) [reply rs/nns/gtc/canister/canister.rs:182]: I was supporting this change. I pointed this out, because it is an exception to my general "these seem like tangential comment reformat changes" feedback..

File: rs/nervous_system/common/build_metadata/src/lib.rs (hunk 6 above)
Comment (daniel-wong-dfinity-org) [reply rs/nervous_system/common/build_metadata/src/lib.rs:36]: I did not realize these comment changes satisfy clippy.

**Review by daniel-wong-dfinity-org — APPROVED** [review]

File: rs/boundary_node/ic_boundary/src/acme.rs (hunk 5 above)
Comment (DSharifi) [reply rs/boundary_node/ic_boundary/src/acme.rs:36]: Do you mind if the @dfinity/boundary-node team rather creates a ticket and cleans up the code you own in a follow up MR, to not add more noise to the already gigantic MR?

File: rs/nns/governance/src/neuron_store.rs (hunk 7 above)
Comment (DSharifi) [reply rs/nns/governance/src/neuron_store.rs:112]: Sure. But I have received comments going the other way, where I commented out dead code or added a `allow` statement, but was requested to delete it.

If the code is unused I will assume it dead code unless someone points it out during a code review.

**Review by basvandijk — APPROVED** [review]

File: rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs [hunk 9]
```diff
@@ -941,14 +941,14 @@ impl EccPoint {
 
     /// Constant-time multiscalar multiplication using Pippenger's algorithm
     ///
-    /// Return point_scalar_pairs[1].0 * point_scalar_pairs[1].1 + ...
-    /// + point_scalar_pairs[n].0 * point_scalar_pairs[n].1
+    /// Return
+    /// point_scalar_pairs[1].0 * point_scalar_pairs[1].1 + point_scalar_pairs[n].0 * point_scalar_pairs[n].1
```
Comment (randombit) [inline rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs:945]: This seems wrong, removes the `...` to indicate a range

Comment (randombit) [reply rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs:945]: (Also the starting index should be 0! But that was a bug in the original comment)

File: rs/consensus/src/consensus.rs (hunk 1 above)
Comment (kpop-dfinity) [reply rs/consensus/src/consensus.rs:380]: Ouch!

File: rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs (hunk 9 above)
Comment (DSharifi) [reply rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs:945]:
```suggestion
    /// point_scalar_pairs[0].0 * point_scalar_pairs[1].1 + ...
    /// \+ point_scalar_pairs[n].0 * point_scalar_pairs[n].1
```

Does this look better?

Comment (DSharifi) [reply rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs:945]: The escape `\` is needed, otherwise the MD generator will interpret the `+` as a bullet point.

Comment (randombit) [reply rs/crypto/internal/crypto_lib/threshold_sig/tecdsa/src/utils/group.rs:945]: LGTM thank you!

**Review by Bownairo — APPROVED** [review]

**Review by THLO — APPROVED** [review]

File: rs/boundary_node/ic_boundary/src/acme.rs (hunk 5 above)
Comment (DSharifi) [reply rs/boundary_node/ic_boundary/src/acme.rs:36]: We agreed to fix this in follow up PRs
